src/Parser/DeathKnight/Unholy/Modules/Features/DarkTransformationAndWounds.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+import Wrapper from 'common/Wrapper';
+
+class DarkTransformationAndWounds extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  // used to track how many stacks a target has
+  targets = {};
+
+  darkTransformationActive = 0;
+  darkTransformationCasts = 0;
+  stopWatchActive = 0;
+  stopWatch = 0;
+  totalTime = 0;
+  
+  
+  on_byPlayer_applybuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = 1;
+		  this.darkTransformationCasts += 1;
+	  }
+  }
+  
+  on_byPlayer_removebuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = 0;
+	  }
+  }
+  
+  on_byPlayer_applydebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+	  }
+  }
+
+  on_byPlayer_removedebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+    }
+  }
+
+  on_byPlayer_removedebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = 0;
+	  }
+  }
+
+  on_byPlayer_cast(event){
+    if(this.targets.hasOwnProperty(encodeTargetString(event.targetID, event.targetInstance))) {
```

Comment:
you can have the null check inside the same if statement as the check for 0 stacks below with a logical OR (aslong as the null check is first), since if it's null it means you have 0 stacks on the target aswell.